src/pg4n/psqlwrapper.py

```diff
@@ -47,13 +49,34 @@ def __init__(self, db_name_parameter: bytes,
     def start(self) -> None:
         """Start psql process and then start feeding hook function with \
         intercepted output."""
-        c = pexpect.spawn("psql " + bytes.decode(self.db_name),
+
+        version_msg = self.check_psql_version()
+        if version_msg != "":
+            print(version_msg)
+
+        c = pexpect.spawn("psql " + bytes.decode(self.psql_args),
                           encoding="utf-8",
                           dimensions=(self.rows, self.cols))
 
         c.interact(input_filter=self.ifilter,
                    output_filter=self.ofilter)
 
+    def check_psql_version(self) -> str:
+        version_info = \
+            pexpect.spawn("psql " + bytes.decode(self.psql_args) +
+                          " --version")
+        version_info.expect(pexpect.EOF)
+        version_info_str: str = bytes.decode(version_info.before)
+        version: str = self.parser.parse_psql_version(version_info_str)
+        version_ok: bool = reduce(lambda x, y: True if x else version == y,
```

Comment:
Tekeekö tämä muutakin kuin `version in self.supported_psql_versions`? Olisi helppolukuisempi.

Reply:
Aivan, sellainenkin pythonismi löytyy.